completion/completion.bash

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+_BALENA_COMPLETION_DIR=$(builtin cd "`dirname "${BASH_SOURCE[0]}"`" > /dev/null && pwd)
+source "$_BALENA_COMPLETION_DIR/bash/balena_comp"
```

Comment:
What is the purpose of this `completion.bash` file? (And the `completion.zsh` file.) If `completion.bash` is executed through the [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)), thus creating a child process, for example:
```sh
cd balena-cli
chmod +x ./completion/completion.bash
./completion/completion.bash
```
... then I understand that the `source` line will have no practical effect, because it will only apply to the child process that quickly terminates. In other words, it doesn't work. :-) It works if this file is sourced:
```sh
cd balena-cli
source ./completion/completion.bash
```
But in this case, we could directly source `./completion/bash/balena_comp` instead (which is what the README documents, as amended by this PR) -
```sh
cd balena-cli
source ./completion/bash/balena_comp
```
And I think it is somewhat confusing to end users to have these 2 files:
```sh
completion/completion.bash
completion/bash/balena_comp
```
or
```sh
completion/completion.zsh
completion/bash/_balena
```
Unless there are some specific reasons to do it like you did, I would suggest the following:
* Delete the old (existing before this PR) `balena-completion.bash`
* Delete `completion/completion.bash`
* Delete `completion/completion.zsh`
* Rename/move `completion/bash/balena_comp` to `completion/balena-completion.bash`
* Rename/move `completion/zsh/_balena` to `completion/balena-completion.zsh`
* Delete subfolder `completion/bash`
* Delete subfolder `completion/zsh`

To make it even clearer, I would also move the template files to a template subfolder:
* Rename/move `completion/balena_bash_template` to `completion/templates/bash.template`
* Rename/move `completion/balena_zsh_template` to `completion/templates/zsh.template`

So we would end up with the following structure:
```sh
completion/balena-completion.bash
completion/balena-completion.zsh
completion/generate-completion.js
completion/templates/bash.template
completion/templates/zsh.template
```

Reply:
The completion.bash/.zsh allow to be an interface for completion script. You just need to source them once and further changes on completion will be automatically loaded on update. This also allow to have multiple completion script loaded for user without them to have to manually source them.

but yes for balena-cli, there is only one tools so the interface isn't needed ...
One issue is that zsh is expecting a file starting with _ for completion and we could only add a whole directory to the completion search path

Reply:
I have modify accordingly. Only the zsh completion file name was keeped to _balena as this is needed to work